spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -1806,13 +1843,19 @@ public void handleConsumeOk(String consumerTag) {
 				latch.countDown();
 			}
 
+			@Override
+			public void handleDelivery(String consumerTag, Envelope envelope, BasicProperties properties, byte[] body)
+					throws IOException {
+				future.complete(new Delivery(envelope, properties, body));
+			}
+
 		};
 		channel.basicConsume(queueName, consumer);
 		if (!latch.await(10, TimeUnit.SECONDS)) {
```

Comment:
But looks like we are blocking for 10 secs plus that `receiveTimeout`. At most...
Also I can live with such a distinguishes between warming and real process.

If you see that there also can be some races, let's just leave as is and go ahead!